### PR TITLE
Contrast 18474 update contrast dot net sdk

### PR DESCRIPTION
**Pull request by rduran0**
Updated SDK to handle most recent version of the API:

* Added response classes for all requests methods.
* Updated objects that already existed and most of their field names.
* Removed most deprecated fields.
* Added new urls to NgEndpoints. Deprecated previous Url class Endpoints.
* Deprecated profile request methods.
* Added filters for Server and Trace request methods to specify extra query params.
* Added new request methods to retrieve Trace events data, trace story and trace http request.
* Updated existing tests and created new ones for new methods.
* Updated Program.cs class with new flow. Replace CheckForTrace example with GetTraceByUUID.

## Notes

* Some old fields were left due to being displayed on web UI but not displayed on documentation.
* All unused old fields that weren't removed, were marked as deprecated.

### CODE REVIEW

**Comment by SteveFeldman** [comment]
@rduran0 seems like now is the perfect time to add a Travis job for this eh?

**Review by ldemidov — APPROVED** [review]

File: SampleContrastClient/Program.cs [hunk 1]
```diff
@@ -86,7 +86,7 @@ static void Main(string[] args)
 
                         //foreach (Trace trace in traces)
                         //{
-                        //    Console.WriteLine("Trace Exists:{0}", DoesTraceExistForUrl(client, appId, trace.Request.Uri));
```
Comment (gmoore) [inline SampleContrastClient/Program.cs:89]: Just curious: why are we updating commented code?

File: SampleContrastClient/Program.cs [hunk 2]
```diff
@@ -137,13 +137,12 @@ private static void DownloadAgentToDesktop(TeamServerClient client)
             }
         }
 
-        // Example usage of CheckForTrace method
-        private static bool DoesTraceExistForUrl( TeamServerClient client, string applicationId, string url )
+        // Example usage of GetTracesByUUID method
+        private static bool GetTraceByUUID( TeamServerClient client, string traceUuid, string organizationId )
```
Comment (gmoore) [inline SampleContrastClient/Program.cs]: If we're not returning the actual trace here, let's rename this method. Something `DoesTraceExist`

**Review by gmoore — CHANGES_REQUESTED** [review]
Just a few q's

File: SampleContrastClient/Program.cs (hunk 2 above)
Comment (rduran0) [reply SampleContrastClient/Program.cs]: You are right. I'll change it back.

**Comment by rduran0** [comment]
@SteveFeldman @gmoore I added AppVeyor CI for building this project. I was planing to use Travis at first but then I read in the documents that [Travis builds C# projects with Mono or .NET Runtime in Linux or OS X and it might not be fully compatible with Windows runtimes](https://docs.travis-ci.com/user/languages/csharp/#Build-Environment).

It runs tests also by default. I omitted the deployment part for now since its completely unknown to me what process you follow to deploy it to Nuget.

File: SampleContrastClient/Program.cs (hunk 1 above)
Comment (rduran0) [reply SampleContrastClient/Program.cs:89]: I think it was meant to work as example hence why I updated.

**Review by gmoore — APPROVED** [review]

**Comment by gmoore** [comment]
Do you have a link to the project on AppVeyor?

**Comment by rduran0** [comment]
Sure, here is last run: https://ci.appveyor.com/project/rduran0/contrast-sdk-dotnet

**Pull request merged**